src/components/common/store/componentParamReducer.tsx

```diff
@@ -213,14 +212,19 @@ const flattenToProperties = (
   }
   return result;
 };
+
 // Change the flatten json to xxx=yyy?xxxx=yyyy which can be use in url.
 const formatToUrlParam = (param: ParameterState) => {
   const result = flattenToProperties(param);
   const parts = [];
   for (const key in result) {
     if (Object.prototype.hasOwnProperty.call(result, key)) {
-      // Check if the value is not an empty string before adding to parts
-      if (result[key] !== "") {
+      // Only add the parameter if it's not undefined, null, or an empty string
+      if (
```

Comment:
it seems like here only want a truthy `result[key]`. So the code can just be `if (result[key])` rather than 
```javascript
if (
        result[key] !== undefined &&
        result[key] !== null &&
        result[key] !== ""
      ) 

```

Reply:
That's a good point. These two methods are similar but not exactly the same. In case that we have keys like `isImosDataOnly` equls to false but we still need it appears in url for query(for now we have`isImosOnlyDataset=false`in query param), so I adopted an explicit and conservative way for checking parameters. I will keep this until we find we can abandon all the falsy values and align with ogcapi.

Reply:
👍